### PR TITLE
Proxy classes/attrs to the stub element in SSR scenarios

### LINKED ISSUE #34: Classes/Attrs not proxied upon client side mount
When an element is mounted client side, if it is not yet hydrated, this only renders a `div`, but when the `Lazyhydrate` element has a class applied to it, it will blow away any internal classes from the root slot element.

You can work around this at the moment by just never including classes or attributes on the `LazyHydrate` component, but that's a bit fragile since developers are quite used to classes/attrs being proxied through to the root child element I think (at least in our case this bit us fairly easily).

I created a small reproduction in https://github.com/brophdawg11/vue-ssr-lazy-hydration-classname-bug to show off the issue, and I have a proposed fix I will open in a PR shortly.

Thanks for the great work on this component!  It has helped our performance metrics a ton in our project 👍

### PR DESCRIPTION
**Pull request by brophdawg11**
Fixes #34

### CODE REVIEW

File: src/LazyHydrate.js [hunk 1]
```diff
@@ -240,7 +240,18 @@ export default {
...
-    const vnode = h(tag);
+
+    let data = {};
+
+    // When we are not hydrated client side in an SSR app, make sure we proxy
+    // along classes/attrs from the SSR element.  Fixes #34.
+    if (this.$el) {
+      data.attrs = Array.from(this.$el.attributes)
+          .reduce((acc, { name, value }) => Object.assign(acc, { [name]: value }), {}),
+      data.class = Array.from(this.$el.classList);
```
Comment (brophdawg11) [inline src/LazyHydrate.js]: This seems to cause a suplicate class issue, so in the reproduction example:

```
        <LazyHydrate :triggerHydration="hydrated" class="lazy-class-outer">
            <div class="lazy-class">
                <button @click="onClick">
                    Click me
                </button>
            </div>
        </LazyHydrate>
```

We end up with the following after mount but before hydration:

```
            <div class="lazy-class-outer lazy-class lazy-class-outer">
```

I think it's probably feasible to filter out the dup class using `this.$vnode.data.staticClass`?

Comment (brophdawg11) [reply src/LazyHydrate.js]: Filtered out the `staticClass` in 98afb99

**Comment by maoberlehner** [comment]
Hey @brophdawg11 just letting you know that I've not forgotten about that pull request. I will try to shift my priorities so I can take a closer look at the weekend. Thx!

**Comment by maoberlehner** [comment]
After taking a closer look I'm not sure if this is going in the right direction.

```html
<LazyHydrate :triggerHydration="hydrated" class="lazy-class-outer">
  <div class="lazy-class">
    <button @click="onClick">
      Click me
    </button>
  </div>
</LazyHydrate>
```

I agree that the _renderless_ nature of `<LazyHydrate>` makes this example confusing for somebody who expects `<LazyHydrate>` to render a wrapper `<div>`. But I think, because `<LazyHydrate>` is renderless it makes no sense to apply `lazy-class-outer` onto `<LazyHydrate>`.

In my opinion there are two ways how to deal with it:

1. Ignoring `lazy-class-outer` class.
2. Proxying `lazy-class-outer` to `<div class="lazy-class">`.

Both ways are _wrong_ in my opinion. I think there is no _right_ way how to handle this gracefully.

Why do I think both ways are wrong: as I see it, the only reason why to put a class on `<LazyHydrate>` is because the developer who does it thinks that `<LazyHydrate>` does render a `<div>`.
The expected output in such case is the following:

```html
<div class="lazy-class-outer">
  <div class="lazy-class">
    <button>
      Click me
    </button>
  </div>
</div>
```

But this is not what `<LazyHydrate>` is actually rendering. But a developer who applies `lazy-class-outer` onto the outer element most likely does this because they want to apply CSS styles which may rely on a HTML structure like you see above. Otherwise, again, it makes no sense to apply `lazy-class-outer` onto `<LazyHydrate>` in the first place.

- I think applying a class on `<LazyHydrate>` is an error and should not be encouraged.
- I think there is no _right_ way how to handle this so everybody would get the outcome they'd expect.

Possible solutions for this project:

- Log a console warning when attributes or classes are applied onto `<LazyHydrate>`.
- Add a chapter in the README explaining that attributes and classes do not work on `<LazyHydrate>`.

Possible solutions for your use case:

If the renderless nature is confusing to your team, you can use https://github.com/znck/lazy-hydration It works pretty much the same as this package (in fact most of my code is based on the work of @znck) with the exception that this `lazy-hydration` is **not** renderless and renders a wrapper `<div>` which means applying classes on it should work as expected (although I've not tested it).

Thanks for your contribution!

**Comment by brophdawg11** [comment]
Thank you for the thorough response.  I see your point and it makes sense when viewing `LazyHydrate` as a true renderless component that effectively disappears upon hydration.  So I agree that both of the solutions, when viewing it this way do feel wrong.

I think both of your solutions make sense, logging a console warning when LazyHydrate is being used incorrectly, and adding a small section to the README.  I can update this PR with those 👍

**Comment by brophdawg11** [comment]
@maoberlehner Updated with a console warning and a new section in the README 👍

**Pull request closed without merging**